File: src/C#/BehavioralPatterns/Mediator/ChainReaction/Dropdown.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ChainReaction
{
    public class Dropdown : Participant
    {
        private readonly Dictionary<string, bool> dropdownItems;

        public Dropdown(Mediator mediator) : base(mediator)
        {
            dropdownItems = new Dictionary<string, bool>()
            {
                { "Auto", false },
                { "Manual", false }
            };
        }

        public string SelectedItem => dropdownItems.FirstOrDefault(x => x.Value).Key;

        public void SelectValue(string key)
        {
            var selectedDropdownItemKey = dropdownItems.FirstOrDefault(x => x.Value).Key;

            if (selectedDropdownItemKey != null)
            {
                dropdownItems[selectedDropdownItemKey] = false;
            }

            dropdownItems[key] = true;
            Console.WriteLine("DropDown value changed to: " + key);
            Mediator.OnStateChanged(this);
        }
    }
}

File: src/PHP/BehavioralPatterns/Mediator/ChainReaction/Dropdown.php
<?php

namespace BehavioralPatterns\Mediator\ChainReaction;

class Dropdown extends Participant
{
    private array $dropdownItems;

    public function __construct(Mediator $mediator)
    {
        parent::__construct($mediator);
        $this->dropdownItems = array(
            "Auto" => false,
            "Manual" => false
        );
    }

    public function getSelectedItem(): ?string
    {
        foreach ($this->dropdownItems as $key => $value) {
            if ($value) {
                return $key;
            }
        }
        return null;
    }

    public function selectValue($key): void
    {
        $selectedDropdownItemKey = $this->getSelectedItem();

        if ($selectedDropdownItemKey !== null) {
            $this->dropdownItems[$selectedDropdownItemKey] = false;
        }

        $this->dropdownItems[$key] = true;
        echo "DropDown value changed to: " . $key . PHP_EOL;
        $this->mediator->onStateChanged($this);
    }
}